<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="serial_port" default="/dev/ttyUSB0" />
    <arg name="baudrate" default="1000000" />
    <arg name="enable_joint_controller" default="true" />
    <arg name="enable_action_controller" default="false" />
    <arg name="publish_servo_state" default="true" />
    <arg name="publish_joint_state" default="true" />
    <arg name="state_publish_freq" default="50" />
    <arg name="respawn" default="true" />
    <arg name="use_sim" default="false" />

    <node name="jetarm_sdk" pkg="jetarm_sdk" type="sdk_node.py" output="screen" respawn="$(arg respawn)">
        <param name="serial_port" type="str" value="$(arg serial_port)" />
        <param name="baudrate" type="int" value="$(arg baudrate)" />
        <param name="use_sim" type="bool" value="$(arg use_sim)" />
        <param name="enable_joint_controller" type="bool" value="$(arg enable_joint_controller)" />
        <param name="publish_servo_state" type="bool" value="$(arg publish_servo_state)" />
        <param name="publish_joint_state" type="bool" value="$(arg publish_joint_state)" />
        <rosparam command="load" file="$(find jetarm_sdk)/config/6dof.yaml" />
    </node>
</launch>
